### ISSUE TITLE
Regression: unit tests fail in CI on MacOS

### ISSUE THREAD

**Issue by oSoMoN** [uutils/diffutils#66]
Last good commit for which all tests passed: e7dc6558c674d07d828a7b4d1024e1dd5749d365.
First bad commit for which unit tests fail on MacOS: d92132e72154a2f59bd053ee523175b7c235049c.

No relevant changes in the codebase itself, this is possibly a regression caused by a dependency, or by a change in the MacOS runners in github.

This is the log of a failed run:
```
  cargo test
  shell: /bin/bash -e {0}
  env:
    CARGO_TERM_COLOR: always
    CARGO_INCREMENTAL: 0
    Updating crates.io index
 Downloading crates ...
  Downloaded anstyle v1.0.6
  Downloaded autocfg v1.1.0
  Downloaded termtree v0.4.1
  Downloaded predicates-tree v1.0.9
  Downloaded normalize-line-endings v0.3.0
  Downloaded float-cmp v0.9.0
  Downloaded wait-timeout v0.2.0
  Downloaded yansi v0.5.1
  Downloaded iana-time-zone v0.1.60
  Downloaded unicode-width v0.1.12
  Downloaded num-traits v0.2.18
  Downloaded serde v1.0.197
  Downloaded rustix v0.38.31
  Downloaded regex-automata v0.4.5
  Downloaded libc v0.2.153
  Downloaded regex-syntax v0.8.2
  Downloaded regex v1.10.4
  Downloaded bstr v1.9.0
  Downloaded pretty_assertions v1.4.0
  Downloaded memchr v2.7.1
  Downloaded chrono v0.4.38
  Downloaded aho-corasick v1.1.2
  Downloaded fastrand v2.0.1
  Downloaded tempfile v3.10.1
  Downloaded diff v0.1.13
  Downloaded same-file v1.0.6
  Downloaded predicates-core v1.0.6
  Downloaded predicates v3.1.0
  Downloaded errno v0.3.8
  Downloaded difflib v0.4.0
  Downloaded core-foundation-sys v0.8.6
  Downloaded cfg-if v1.0.0
  Downloaded bitflags v2.4.2
  Downloaded assert_cmd v2.0.14
  Downloaded doc-comment v0.3.3
   Compiling autocfg v1.1.0
   Compiling memchr v2.7.1
   Compiling libc v0.2.153
   Compiling aho-corasick v1.1.2
   Compiling num-traits v0.2.18
   Compiling regex-syntax v0.8.2
   Compiling core-foundation-sys v0.8.6
   Compiling rustix v0.38.31
   Compiling iana-time-zone v0.1.60
   Compiling doc-comment v0.3.3
   Compiling diff v0.1.13
   Compiling predicates-core v1.0.6
   Compiling regex-automata v0.4.5
   Compiling chrono v0.4.38
   Compiling float-cmp v0.9.0
   Compiling errno v0.3.8
   Compiling anstyle v1.0.6
   Compiling difflib v0.4.0
   Compiling unicode-width v0.1.12
   Compiling assert_cmd v2.0.14
   Compiling bitflags v2.4.2
   Compiling regex v1.10.4
   Compiling same-file v1.0.6
   Compiling normalize-line-endings v0.3.0
   Compiling termtree v0.4.1
   Compiling predicates v3.1.0
   Compiling predicates-tree v1.0.9
   Compiling bstr v1.9.0
   Compiling wait-timeout v0.2.0
   Compiling cfg-if v1.0.0
   Compiling fastrand v2.0.1
   Compiling yansi v0.5.1
   Compiling pretty_assertions v1.4.0
   Compiling tempfile v3.10.1
   Compiling diffutils v0.4.1 (/Users/runner/work/diffutils/diffutils)
    Finished test [unoptimized + debuginfo] target(s) in 15.14s
     Running unittests src/lib.rs (target/debug/deps/diffutilslib-41e658c8501c1bc7)

running 45 tests
test context_diff::tests::test_permutations_missing_lines ... FAILED
test context_diff::tests::test_permutations ... FAILED
test context_diff::tests::test_stop_early ... ok
test ed_diff::tests::test_basic ... ok
test context_diff::tests::test_permutations_empty_lines ... FAILED
test context_diff::tests::test_permutations_reverse ... FAILED
test ed_diff::tests::test_permutations ... ok
test ed_diff::tests::test_stop_early ... ok
test normal_diff::tests::test_basic ... ok
test ed_diff::tests::test_permutations_empty_lines ... ok
test ed_diff::tests::test_permutations_reverse ... ok
test normal_diff::tests::test_permutations_empty_lines ... ok
test normal_diff::tests::test_permutations ... ok
test normal_diff::tests::test_stop_early ... ok
test params::tests::basics ... ok
test params::tests::basics_ed ... ok
test params::tests::brief ... ok
test params::tests::conflicting_output_styles ... ok
test params::tests::context_count ... ok
test params::tests::context_invalid ... ok
test params::tests::context_valid ... ok
test params::tests::default_to_stdin ... ok
test params::tests::double_dash ... ok
test params::tests::empty ... ok
test params::tests::expand_tabs ... ok
test params::tests::missing_arguments ... ok
test params::tests::report_identical_files ... ok
test params::tests::tabsize ... ok
test params::tests::unified_invalid ... ok
test params::tests::unified_valid ... ok
test params::tests::unknown_argument ... ok
test unified_diff::tests::test_permutations ... FAILED
test unified_diff::tests::test_permutations_empty_lines ... FAILED
test unified_diff::tests::test_permutations_missing_line_ending ... FAILED
test unified_diff::tests::test_permutations_missing_lines ... FAILED
test unified_diff::tests::test_permutations_reverse ... FAILED
test unified_diff::tests::test_stop_early ... ok
test utils::tests::expand_tabs::basics ... ok
test utils::tests::expand_tabs::invalid_utf8 ... ok
test utils::tests::expand_tabs::multibyte_chars ... ok
test utils::tests::modification_time::invalid_file ... ok
test utils::tests::modification_time::set_time ... ok
test utils::tests::write_line::basics ... ok
test normal_diff::tests::test_permutations_reverse ... ok
test normal_diff::tests::test_permutations_missing_line_ending ... ok

failures:

---- context_diff::tests::test_permutations_missing_lines stdout ----
thread 'context_diff::tests::test_permutations_missing_lines' panicked at src/context_diff.rs:627:33:
assertion failed: `(left == right)`

Diff < left / right > :
<[]
>[
>    98,
>    10,
>    100,
>    10,
>    102,
>    10,
>    104,
>    10,
>    106,
>    10,
>    108,
>    10,
>]


note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace

---- context_diff::tests::test_permutations stdout ----
thread 'context_diff::tests::test_permutations' panicked at src/context_diff.rs:462:33:
assertion failed: `(left == right)`

error: test failed, to rerun pass `--lib`
Diff < left / right > :
 [
     100,
     10,
<    102,
<    10,
<    104,
<    10,
<    106,
<    10,
<    108,
<    10,
 ]



---- context_diff::tests::test_permutations_empty_lines stdout ----
thread 'context_diff::tests::test_permutations_empty_lines' panicked at src/context_diff.rs:543:33:
assertion failed: `(left == right)`

Diff < left / right > :
 [
     100,
     10,
<    102,
<    10,
<    104,
<    10,
<    106,
<    10,
<    108,
<    10,
 ]



---- context_diff::tests::test_permutations_reverse stdout ----
thread 'context_diff::tests::test_permutations_reverse' panicked at src/context_diff.rs:714:33:
assertion failed: `(left == right)`

Diff < left / right > :
 [
     98,
     10,
<    100,
<    10,
<    99,
<    10,
<    98,
<    10,
<    97,
<    10,
 ]



---- unified_diff::tests::test_permutations stdout ----
diff: "--- a/alef\t2024-04-29 16:15:16.214785000 +0000\n+++ target/unified-diff//alef\t2024-04-29 16:15:16.215087000 +0000\n@@ -1,6 +1,6 @@\n-a\n-c\n-e\n-g\n-i\n-k\n+b\n+d\n+f\n+h\n+j\n+l\n"
alef: "a\nc\ne\ng\ni\nk\n"
bet: "b\nd\nf\nh\nj\nl\n"
patching file 'target/unified-diff//alef'


diff: "--- a/alef\t2024-04-29 16:15:16.253414000 +0000\n+++ target/unified-diff//alef\t2024-04-29 16:15:16.226020177 +0000\n@@ -1,6 +1,6 @@\n-a\n-c\n-e\n-g\n-i\n+b\n+d\n+f\n+h\n+j\n l\n"
alef: "a\nc\ne\ng\ni\nl\n"
bet: "b\nd\nf\nh\nj\nl\n"
patching file 'target/unified-diff//alef'


diff: "--- a/alef\t2024-04-29 16:15:16.259269000 +0000\n+++ target/unified-diff//alef\t2024-04-29 16:15:16.258911885 +0000\n@@ -1,6 +1,5 @@\n-a\n-c\n-e\n-g\n-i\n-l\n+b\n+d\n+f\n+h\n+j\n"
alef: "a\nc\ne\ng\ni\nl\n"
bet: "b\nd\nf\nh\nj\n"
patching file 'target/unified-diff//alef'


diff: "--- a/alef\t2024-04-29 16:15:16.292388000 +0000\n+++ target/unified-diff//alef\t2024-04-29 16:15:16.291242718 +0000\n@@ -1,6 +1,6 @@\n-a\n-c\n-e\n-g\n+b\n+d\n+f\n+h\n j\n-k\n+l\n"
alef: "a\nc\ne\ng\nj\nk\n"
bet: "b\nd\nf\nh\nj\nl\n"
patching file 'target/unified-diff//alef'


diff: "--- a/alef\t2024-04-29 16:15:16.324557000 +0000\n+++ target/unified-diff//alef\t2024-04-29 16:15:16.323348968 +0000\n@@ -1,6 +1,6 @@\n-a\n-c\n-e\n-g\n+b\n+d\n+f\n+h\n j\n l\n"
alef: "a\nc\ne\ng\nj\nl\n"
bet: "b\nd\nf\nh\nj\nl\n"
patching file 'target/unified-diff//alef'


diff: "--- a/alef\t2024-04-29 16:15:16.354967000 +0000\n+++ target/unified-diff//alef\t2024-04-29 16:15:16.353538177 +0000\n@@ -1,6 +1,5 @@\n-a\n-c\n-e\n-g\n+b\n+d\n+f\n+h\n j\n-l\n"
alef: "a\nc\ne\ng\nj\nl\n"
bet: "b\nd\nf\nh\nj\n"
patching file 'target/unified-diff//alef'


diff: "--- a/alef\t2024-04-29 16:15:16.388010000 +0000\n+++ target/unified-diff//alef\t2024-04-29 16:15:16.387420177 +0000\n@@ -1,6 +1,5 @@\n-a\n-c\n-e\n-g\n-j\n-k\n+b\n+d\n+f\n+h\n+l\n"
alef: "a\nc\ne\ng\nj\nk\n"
bet: "b\nd\nf\nh\nl\n"
patching file 'target/unified-diff//alef'


diff: "--- a/alef\t2024-04-29 16:15:16.423191000 +0000\n+++ target/unified-diff//alef\t2024-04-29 16:15:16.422733635 +0000\n@@ -1,6 +1,5 @@\n-a\n-c\n-e\n-g\n-j\n+b\n+d\n+f\n+h\n l\n"
alef: "a\nc\ne\ng\nj\nl\n"
bet: "b\nd\nf\nh\nl\n"
patching file 'target/unified-diff//alef'


diff: "--- a/alef\t2024-04-29 16:15:16.455595000 +0000\n+++ target/unified-diff//alef\t2024-04-29 16:15:16.439889052 +0000\n@@ -1,6 +1,4 @@\n-a\n-c\n-e\n-g\n-j\n-l\n+b\n+d\n+f\n+h\n"
alef: "a\nc\ne\ng\nj\nl\n"
bet: "b\nd\nf\nh\n"
patching file 'target/unified-diff//alef'


diff: "--- a/alef\t2024-04-29 16:15:16.461551000 +0000\n+++ target/unified-diff//alef\t2024-04-29 16:15:16.461122677 +0000\n@@ -1,6 +1,6 @@\n-a\n-c\n-e\n+b\n+d\n+f\n h\n-i\n-k\n+j\n+l\n"
alef: "a\nc\ne\nh\ni\nk\n"
bet: "b\nd\nf\nh\nj\nl\n"
patching file 'target/unified-diff//alef'


diff: "--- a/alef\t2024-04-29 16:15:16.490883000 +0000\n+++ target/unified-diff//alef\t2024-04-29 16:15:16.482739718 +0000\n@@ -1,6 +1,6 @@\n-a\n-c\n-e\n+b\n+d\n+f\n h\n-i\n+j\n l\n"
alef: "a\nc\ne\nh\ni\nl\n"
bet: "b\nd\nf\nh\nj\nl\n"
patching file 'target/unified-diff//alef'


diff: "--- a/alef\t2024-04-29 16:15:16.520231000 +0000\n+++ target/unified-diff//alef\t2024-04-29 16:15:16.513848468 +0000\n@@ -1,6 +1,5 @@\n-a\n-c\n-e\n+b\n+d\n+f\n h\n-i\n-l\n+j\n"
alef: "a\nc\ne\nh\ni\nl\n"
bet: "b\nd\nf\nh\nj\n"
patching file 'target/unified-diff//alef'


diff: "--- a/alef\t2024-04-29 16:15:16.529413000 +0000\n+++ target/unified-diff//alef\t2024-04-29 16:15:16.528755093 +0000\n@@ -1,5 +1,5 @@\n-a\n-c\n-e\n+b\n+d\n+f\n h\n j\n@@ -6 +6 @@\n-k\n+l\n"
alef: "a\nc\ne\nh\nj\nk\n"
bet: "b\nd\nf\nh\nj\nl\n"
patching file 'target/unified-diff//alef'


diff: "--- a/alef\t2024-04-29 16:15:16.555854000 +0000\n+++ target/unified-diff//alef\t2024-04-29 16:15:16.555106385 +0000\n@@ -1,5 +1,5 @@\n-a\n-c\n-e\n+b\n+d\n+f\n h\n j\n"
alef: "a\nc\ne\nh\nj\nl\n"
bet: "b\nd\nf\nh\nj\nl\n"
patching file 'target/unified-diff//alef'


diff: "--- a/alef\t2024-04-29 16:15:16.560091000 +0000\n+++ target/unified-diff//alef\t2024-04-29 16:15:16.558862718 +0000\n@@ -1,5 +1,5 @@\n-a\n-c\n-e\n+b\n+d\n+f\n h\n j\n@@ -6 +5,0 @@\n-l\n"
alef: "a\nc\ne\nh\nj\nl\n"
bet: "b\nd\nf\nh\nj\n"
patching file 'target/unified-diff//alef'


diff: "--- a/alef\t2024-04-29 16:15:16.567411000 +0000\n+++ target/unified-diff//alef\t2024-04-29 16:15:16.566992010 +0000\n@@ -1,6 +1,5 @@\n-a\n-c\n-e\n+b\n+d\n+f\n h\n-j\n-k\n+l\n"
alef: "a\nc\ne\nh\nj\nk\n"
bet: "b\nd\nf\nh\nl\n"
patching file 'target/unified-diff//alef'


diff: "--- a/alef\t2024-04-29 16:15:16.638089000 +0000\n+++ target/unified-diff//alef\t2024-04-29 16:15:16.583116510 +0000\n@@ -1,6 +1,5 @@\n-a\n-c\n-e\n+b\n+d\n+f\n h\n-j\n l\n"
alef: "a\nc\ne\nh\nj\nl\n"
bet: "b\nd\nf\nh\nl\n"
patching file 'target/unified-diff//alef'


diff: "--- a/alef\t2024-04-29 16:15:16.649305000 +0000\n+++ target/unified-diff//alef\t2024-04-29 16:15:16.648936218 +0000\n@@ -1,6 +1,4 @@\n-a\n-c\n-e\n+b\n+d\n+f\n h\n-j\n-l\n"
alef: "a\nc\ne\nh\nj\nl\n"
bet: "b\nd\nf\nh\n"
patching file 'target/unified-diff//alef'


diff: "--- a/alef\t2024-04-29 16:15:16.719642000 +0000\n+++ target/unified-diff//alef\t2024-04-29 16:15:16.671017677 +0000\n@@ -1,6 +1,5 @@\n-a\n-c\n-e\n-h\n-i\n-k\n+b\n+d\n+f\n+j\n+l\n"
alef: "a\nc\ne\nh\ni\nk\n"
bet: "b\nd\nf\nj\nl\n"
patching file 'target/unified-diff//alef'


diff: "--- a/alef\t2024-04-29 16:15:16.727342000 +0000\n+++ target/unified-diff//alef\t2024-04-29 16:15:16.725870718 +0000\n@@ -1,6 +1,5 @@\n-a\n-c\n-e\n-h\n-i\n+b\n+d\n+f\n+j\n l\n"
alef: "a\nc\ne\nh\ni\nl\n"
bet: "b\nd\nf\nj\nl\n"
patching file 'target/unified-diff//alef'


diff: "--- a/alef\t2024-04-29 16:15:16.736861000 +0000\n+++ target/unified-diff//alef\t2024-04-29 16:15:16.735508552 +0000\n@@ -1,6 +1,4 @@\n-a\n-c\n-e\n-h\n-i\n-l\n+b\n+d\n+f\n+j\n"
alef: "a\nc\ne\nh\ni\nl\n"
bet: "b\nd\nf\nj\n"
patching file 'target/unified-diff//alef'


diff: "--- a/alef\t2024-04-29 16:15:16.789218000 +0000\n+++ target/unified-diff//alef\t2024-04-29 16:15:16.747693927 +0000\n@@ -1,6 +1,5 @@\n-a\n-c\n-e\n-h\n+b\n+d\n+f\n j\n-k\n+l\n"
alef: "a\nc\ne\nh\nj\nk\n"
bet: "b\nd\nf\nj\nl\n"
patching file 'target/unified-diff//alef'


diff: "--- a/alef\t2024-04-29 16:15:16.829124000 +0000\n+++ target/unified-diff//alef\t2024-04-29 16:15:16.827835427 +0000\n@@ -1,6 +1,5 @@\n-a\n-c\n-e\n-h\n+b\n+d\n+f\n j\n l\n"
alef: "a\nc\ne\nh\nj\nl\n"
bet: "b\nd\nf\nj\nl\n"
patching file 'target/unified-diff//alef'


diff: "--- a/alef\t2024-04-29 16:15:16.834407000 +0000\n+++ target/unified-diff//alef\t2024-04-29 16:15:16.833906218 +0000\n@@ -1,6 +1,4 @@\n-a\n-c\n-e\n-h\n+b\n+d\n+f\n j\n-l\n"
alef: "a\nc\ne\nh\nj\nl\n"
bet: "b\nd\nf\nj\n"
patching file 'target/unified-diff//alef'


diff: "--- a/alef\t2024-04-29 16:15:16.866112000 +0000\n+++ target/unified-diff//alef\t2024-04-29 16:15:16.859225968 +0000\n@@ -1,6 +1,4 @@\n-a\n-c\n-e\n-h\n-j\n-k\n+b\n+d\n+f\n+l\n"
alef: "a\nc\ne\nh\nj\nk\n"
bet: "b\nd\nf\nl\n"
patching file 'target/unified-diff//alef'


diff: "--- a/alef\t2024-04-29 16:15:16.873121000 +0000\n+++ target/unified-diff//alef\t2024-04-29 16:15:16.872291510 +0000\n@@ -1,6 +1,4 @@\n-a\n-c\n-e\n-h\n-j\n+b\n+d\n+f\n l\n"
alef: "a\nc\ne\nh\nj\nl\n"
bet: "b\nd\nf\nl\n"
patching file 'target/unified-diff//alef'


diff: "--- a/alef\t2024-04-29 16:15:16.904452000 +0000\n+++ target/unified-diff//alef\t2024-04-29 16:15:16.898056885 +0000\n@@ -1,6 +1,3 @@\n-a\n-c\n-e\n-h\n-j\n-l\n+b\n+d\n+f\n"
alef: "a\nc\ne\nh\nj\nl\n"
bet: "b\nd\nf\n"
patching file 'target/unified-diff//alef'


diff: "--- a/alef\t2024-04-29 16:15:16.913479000 +0000\n+++ target/unified-diff//alef\t2024-04-29 16:15:16.911049343 +0000\n@@ -1,6 +1,6 @@\n-a\n-c\n+b\n+d\n f\n-g\n-i\n-k\n+h\n+j\n+l\n"
alef: "a\nc\nf\ng\ni\nk\n"
bet: "b\nd\nf\nh\nj\nl\n"
patching file 'target/unified-diff//alef'


diff: "--- a/alef\t2024-04-29 16:15:16.920496000 +0000\n+++ target/unified-diff//alef\t2024-04-29 16:15:16.919117802 +0000\n@@ -1,6 +1,6 @@\n-a\n-c\n+b\n+d\n f\n-g\n-i\n+h\n+j\n l\n"
alef: "a\nc\nf\ng\ni\nl\n"
bet: "b\nd\nf\nh\nj\nl\n"
patching file 'target/unified-diff//alef'


diff: "--- a/alef\t2024-04-29 16:15:16.996868000 +0000\n+++ target/unified-diff//alef\t2024-04-29 16:15:16.941167468 +0000\n@@ -1,6 +1,5 @@\n-a\n-c\n+b\n+d\n f\n-g\n-i\n-l\n+h\n+j\n"
alef: "a\nc\nf\ng\ni\nl\n"
bet: "b\nd\nf\nh\nj\n"
patching file 'target/unified-diff//alef'


diff: "--- a/alef\t2024-04-29 16:15:17.001643000 +0000\n+++ target/unified-diff//alef\t2024-04-29 16:15:17.001244260 +0000\n@@ -1,6 +1,6 @@\n-a\n-c\n+b\n+d\n f\n-g\n+h\n j\n-k\n+l\n"
alef: "a\nc\nf\ng\nj\nk\n"
bet: "b\nd\nf\nh\nj\nl\n"
patching file 'target/unified-diff//alef'


diff: "--- a/alef\t2024-04-29 16:15:17.033890000 +0000\n+++ target/unified-diff//alef\t2024-04-29 16:15:17.020385927 +0000\n@@ -1,6 +1,6 @@\n-a\n-c\n+b\n+d\n f\n-g\n+h\n j\n l\n"
alef: "a\nc\nf\ng\nj\nl\n"
bet: "b\nd\nf\nh\nj\nl\n"
patching file 'target/unified-diff//alef'


diff: "--- a/alef\t2024-04-29 16:15:17.041841000 +0000\n+++ target/unified-diff//alef\t2024-04-29 16:15:17.040688468 +0000\n@@ -1,6 +1,5 @@\n-a\n-c\n+b\n+d\n f\n-g\n+h\n j\n-l\n"
alef: "a\nc\nf\ng\nj\nl\n"
bet: "b\nd\nf\nh\nj\n"
patching file 'target/unified-diff//alef'


diff: "--- a/alef\t2024-04-29 16:15:17.047762000 +0000\n+++ target/unified-diff//alef\t2024-04-29 16:15:17.046999093 +0000\n@@ -1,6 +1,5 @@\n-a\n-c\n+b\n+d\n f\n-g\n-j\n-k\n+h\n+l\n"
alef: "a\nc\nf\ng\nj\nk\n"
bet: "b\nd\nf\nh\nl\n"
patching file 'target/unified-diff//alef'


diff: "--- a/alef\t2024-04-29 16:15:17.078825000 +0000\n+++ target/unified-diff//alef\t2024-04-29 16:15:17.069421927 +0000\n@@ -1,6 +1,5 @@\n-a\n-c\n+b\n+d\n f\n-g\n-j\n+h\n l\n"
alef: "a\nc\nf\ng\nj\nl\n"
bet: "b\nd\nf\nh\nl\n"
patching file 'target/unified-diff//alef'


diff: "--- a/alef\t2024-04-29 16:15:17.085486000 +0000\n+++ target/unified-diff//alef\t2024-04-29 16:15:17.084414427 +0000\n@@ -1,6 +1,4 @@\n-a\n-c\n+b\n+d\n f\n-g\n-j\n-l\n+h\n"
alef: "a\nc\nf\ng\nj\nl\n"
bet: "b\nd\nf\nh\n"
patching file 'target/unified-diff//alef'


diff: "--- a/alef\t2024-04-29 16:15:17.123642000 +0000\n+++ target/unified-diff//alef\t2024-04-29 16:15:17.097894093 +0000\n@@ -1,4 +1,4 @@\n-a\n-c\n+b\n+d\n f\n h\n@@ -5,2 +5,2 @@\n-i\n-k\n+j\n+l\n"
alef: "a\nc\nf\nh\ni\nk\n"
bet: "b\nd\nf\nh\nj\nl\n"
patching file 'target/unified-diff//alef'


diff: "--- a/alef\t2024-04-29 16:15:17.133596000 +0000\n+++ target/unified-diff//alef\t2024-04-29 16:15:17.132631802 +0000\n@@ -1,4 +1,4 @@\n-a\n-c\n+b\n+d\n f\n h\n@@ -5,2 +5,2 @@\n-i\n+j\n l\n"
alef: "a\nc\nf\nh\ni\nl\n"
bet: "b\nd\nf\nh\nj\nl\n"
patching file 'target/unified-diff//alef'
1 out of 2 hunks failed--saving rejects to 'target/unified-diff//alef.rej'


thread 'unified_diff::tests::test_permutations' panicked at src/unified_diff.rs:502:33:
Output { status: ExitStatus(unix_wait_status(256)), stdout: "patching file 'target/unified-diff//alef'\n1 out of 2 hunks failed--saving rejects to 'target/unified-diff//alef.rej'\n", stderr: "" }

---- unified_diff::tests::test_permutations_empty_lines stdout ----
thread 'unified_diff::tests::test_permutations_empty_lines' panicked at src/unified_diff.rs:696:37:
Output { status: ExitStatus(unix_wait_status(256)), stdout: "patching file 'target/unified-diff//alef_'\n1 out of 2 hunks failed--saving rejects to 'target/unified-diff//alef_.rej'\n", stderr: "" }

---- unified_diff::tests::test_permutations_missing_line_ending stdout ----
thread 'unified_diff::tests::test_permutations_missing_line_ending' panicked at src/unified_diff.rs:600:37:
Output { status: ExitStatus(unix_wait_status(256)), stdout: "patching file 'target/unified-diff//alefn'\n1 out of 2 hunks failed--saving rejects to 'target/unified-diff//alefn.rej'\n", stderr: "" }

---- unified_diff::tests::test_permutations_missing_lines stdout ----
thread 'unified_diff::tests::test_permutations_missing_lines' panicked at src/unified_diff.rs:781:33:
assertion failed: `(left == right)`

Diff < left / right > :
 [
<    97,
>    98,
     10,
<    99,
>    100,
     10,
<    101,
>    102,
     10,
<    103,
>    104,
     10,
<    105,
>    106,
     10,
<    107,
>    108,
     10,
 ]



---- unified_diff::tests::test_permutations_reverse stdout ----
thread 'unified_diff::tests::test_permutations_reverse' panicked at src/unified_diff.rs:863:33:
Output { status: ExitStatus(unix_wait_status(256)), stdout: "patching file 'target/unified-diff//alefr'\n1 out of 2 hunks failed--saving rejects to 'target/unified-diff//alefr.rej'\n", stderr: "" }


failures:
    context_diff::tests::test_permutations
    context_diff::tests::test_permutations_empty_lines
    context_diff::tests::test_permutations_missing_lines
    context_diff::tests::test_permutations_reverse
    unified_diff::tests::test_permutations
    unified_diff::tests::test_permutations_empty_lines
    unified_diff::tests::test_permutations_missing_line_ending
    unified_diff::tests::test_permutations_missing_lines
    unified_diff::tests::test_permutations_reverse

test result: FAILED. 36 passed; 9 failed; 0 ignored; 0 measured; 0 filtered out; finished in 69.33s

Error: Process completed with exit code 101.
```

**Comment by oSoMoN** [comment]
In the logs of the last successful run and the first failed run, I'm not seeing any differences in the versions of the dependencies being pulled from crates.io. In particular, the change isn't caused by #65. So I'm leaning towards a change in the github macOS runners.

**Comment by oSoMoN** [comment]
The tests that fail all run the `patch` binary, so I initially suspected that maybe the binary had been removed from the default macOS image in github runners. However I verified that it is still there, installed at `/usr/bin/patch`.

The output of `patch --version` however is quite different from what's seen on a Linux runner:
```
patch --version
patch 2.0-12u11-Apple
```
(to be compared to GNU patch 2.7.6 on Linux)

**Comment by oSoMoN** [comment]
Installing GNU patch on MacOS runners (with `brew install gpatch`) does the trick. That looks like a regression/change in the default image for MacOS runners, indeed.

**Comment by oSoMoN** [comment]
This had regressed a few months ago, and I fixed it again with 44c195c0b22f653f9f3d0ad89f789df06b35dbb5.
